frontend/server/src/Controllers/Course.php

```diff
@@ -3277,6 +3344,118 @@ private static function getFilteredCourses($courses, $coursesTypes) {
         return $filteredCourses;
     }
 
+    /**
+     * @param array{userRegistrationAccepted?: bool|null, userRegistrationAnswered: bool, userRegistrationRequested: bool}|array<empty, empty> $registrationResponse
+     *
+     * @return array{entrypoint: string, smartyProperties: array{coursePayload: IntroDetailsPayload, payload: IntroDetailsPayload, title: \OmegaUp\TranslationString}}
+     */
+    private static function getIntroDetailsForCourse(
+        \OmegaUp\DAO\VO\Courses $course,
+        ?\OmegaUp\DAO\VO\Identities $identity = null,
+        bool $shouldShowIntro = true,
+        ?bool $hasAcceptedTeacher = true,
```

Comment:
```suggestion
        bool $hasAcceptedTeacher = true,
```

no tiene mucho sentido que este booleano sea nullable. sobre todo porque al final sólo nos interesa que no sea `true`, entonces si era nulo al momento de llamar esta función podemos hacer `?? false`.

Reply:
Ok, se deja el parámetro como no-opcional